File: Medium/633_Sum_of_Square_Numbers/Solution.py
from typing import List, Optional
from math import sqrt


class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        left = 0
        right = int(sqrt(c))

        while left <= right:
            sum = left * left + right * right
            if sum == c:
                return True
            elif sum < c:
                left += 1
            else:
                right -= 1
        
        return False


if __name__ == "__main__":
    solution = Solution()

    # test cases 1
    print(solution.judgeSquareSum(5))   # true
    
    # test cases 2
    print(solution.judgeSquareSum(3))   # false

File: Medium/633_Sum_of_Square_Numbers/Solution.java



public class Solution {
    public boolean judgeSquareSum(int c) {
        long left = 0;
        long right = (long) Math.sqrt(c);
        
        while (left <= right) {
            long sum = left * left + right * right;
            if (sum == c) return true;
            if (sum < c) left++;
            else right--;
        }
        
        return false;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        System.out.println(solution.judgeSquareSum(5));   // true
        
        // test cases 2
        System.out.println(solution.judgeSquareSum(3));   // false
    }
}